avro-converter/src/main/java/io/confluent/connect/avro/AvroData.java

```diff
@@ -879,11 +879,15 @@ private Object toConnectData(Schema schema, Object value) {
         case ARRAY: {
           Schema valueSchema = schema.valueSchema();
           Collection<Object> original = (Collection<Object>) value;
-          List<Object> result = new ArrayList<>(original.size());
-          for (Object elem : original) {
-            result.add(toConnectData(valueSchema, elem));
+          if (original == null) {
```

Comment:
There are lots of tests in `AvroDataTest` to try to cover the various types of data we might try to convert. Can we add some tests to validate these changes? It would also help me understand in which cases these get triggered.

For example, it looks like these checks for null values are important when we have optional arrays/maps/structs nested within another container type since that's the only time this method is called recursively and we wouldn't have hit the `if (value == null)` check in `toConnectData(org.apache.avro.Schema, Object)`. But do we need to do this separately for each type or can we just put a single check at the beginning of the method? It gets hard to reason about all of the different cases we need to handle, so it'd be helpful to have tests for the cases you know were failing to see if we can also simplify the code a bit while still addressing the cases you found that need to be handled differently.